.github/workflows/CI.yml

```diff
@@ -93,7 +92,37 @@ jobs:
       with:
         name: binary_artifacts
         path: build
+        
+  analyze:
+    name: CodeQL Analyze
+    needs: build
+    runs-on: ubuntu-latest
 
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'go' ]
+    
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v1
+        with:
+          languages: ${{ matrix.language }}
+        
+      - name: Download built artifacts
+        uses: actions/download-artifact@v2
+        with:
+          name: binary_artifacts
+          path: build
+
+      - name: Display structure of downloaded files
+        run: ls -R
+
+      - name: Perform CodeQL Analysis
+        uses: github/codeql-action/analyze@v1
```

Comment:
wonder where can we find the result, and can we fail the workflow if the code doesn't match the standard?

Reply:
I believe all alerts should show in the Code scanning tab in Github: https://github.com/open-o11y/aws-otel-collector/security/code-scanning?query=tool%3ACodeQL.

Reply:
i think it would be better if you could add this to the PR-Build workflow as well, so we can identify issue before merging.

Reply:
and can we fail the workflow if there's any alert?

Reply:
It looks like it's currently not possible to fail the CodeQL Github Action: https://github.com/github/codeql-action/issues/304.

Reply:
oh okay, that's fine